### PR TITLE
task: Do not substitute Zed variables in command and env

### LINKED ISSUE #12627: Add a `selected text or current buffer` variable in tasks
### Check for existing issues

- [X] Completed

### Describe the feature

Add a `selected text or current buffer` variable in tasks. If there is no text selected, the value of the variable default to current whole buffer. With this variable, the below two tasks can be merged into one

```
  {
    "label": "Dpaste (File)",
    "command": "cat $ZED_FILE | curl -s -F \"content=<-\" https://dpaste.com/api/v2/ | pbcopy",
    "use_new_terminal": false,
    "allow_concurrent_runs": false,
    "reveal": "never"
  },
  {
    "label": "Dpaste (Region)",
    "command": "echo -e \"$ZED_SELECTED_TEXT\" | curl -s -F \"content=<-\" https://dpaste.com/api/v2/ | pbcopy",
    "use_new_terminal": false,
    "allow_concurrent_runs": false,
    "reveal": "never"
  }
```

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by osiewicz**
/cc @SomeoneToIgnore 
Fixes https://github.com/zed-industries/zed/issues/12627
Release Notes:

- Task variables are no longer substituted in `env` or `command` before handing it off to shell.

### CODE REVIEW

File: crates/task/src/lib.rs [hunk 1]
```diff
@@ -66,8 +66,8 @@ pub struct SpawnInTerminal {
     /// Arguments to the command, potentially unsubstituted,
     /// to let the shell that spawns the command to do the substitution, if needed.
     pub args: Vec<String>,
-    /// A human-readable label, containing command and all of its arguments, joined and substituted.
-    pub command_label: String,
+    /// A command with all of it's arguments appended.
```
Comment (SomeoneToIgnore) [inline crates/task/src/lib.rs:69]:
```suggestion
    /// A command with all of its arguments appended.
```

File: crates/task/src/task_template.rs [hunk 2]
```diff
@@ -143,13 +143,13 @@ impl TaskTemplate {
             &variable_names,
             &mut substituted_variables,
         )?;
-        let command = substitute_all_template_variables_in_str(
+        let _command = substitute_all_template_variables_in_str(
```
Comment (SomeoneToIgnore) [inline crates/task/src/task_template.rs:146]: Can we just remove those?

**Review by SomeoneToIgnore — COMMENTED** [review]
That would mean, we're loosing the 
```
> Task `cargo test -p zed test_new_empty_workspace -- --nocapture` finished successfully
> Command: '/bin/zsh -i -c `cargo test -p zed test_new_empty_workspace -- --nocapture`'
```
expanded terminal outuput due to this?
Can we expand the command for display reasons at least?

File: crates/task/src/task_template.rs (hunk 2 above)
Comment (SomeoneToIgnore) [reply crates/task/src/task_template.rs:146]: Ohh, missed the substitution side effect, sorry.
But still — I think we'd want to reuse the substituted label at least, somewhere.

**Comment by osiewicz** [comment]
Yeah, that'd mean that this `command` would become fully unsubstituted. How'd you feel about just printing the task variables from the context instead?

**Comment by WeetHet** [comment]
I personally think having a fully substituted command outputted is really useful and should not be removed. It really helps to immediately see what was being executed

**Comment by osiewicz** [comment]
I'll close out this PR given unclearness around displaying substituted variables.

**Pull request closed without merging**